cpp/src/arrow/flight/sql/odbc/odbcabstraction/odbc_impl/odbc_descriptor.cc

```diff
@@ -479,8 +479,6 @@ void ODBCDescriptor::PopulateFromResultSetMetadata(ResultSetMetadata* rsmd) {
 
   for (size_t i = 0; i < records_.size(); ++i) {
     size_t one_based_index = i + 1;
-    int16_t concise_type = rsmd->GetConciseType(one_based_index);
```

Comment:
Here we are undoing a mistake from a previous PR. This logic for handling concise type should be handled by a future PR.